beacon_node/store/src/chunked_vector.rs

```diff
@@ -431,9 +431,15 @@ fn range_query<S: KeyValueStore<E>, E: EthSpec, T: Decode + Encode>(
     start_index: usize,
     end_index: usize,
 ) -> Result<Vec<Chunk<T>>, Error> {
-    let mut result = vec![];
+    let range = start_index..=end_index;
+    let len = range
```

Comment:
I guess you were angling for `ExactSizeIterator::len`? I think given that it doesn't exist for `RangeInclusive<usize>` due to overflow we may as well write it manually like:

```suggestion
    let len = end_index.saturating_add(1).saturating_sub(start_index);
```

Reply:
> I guess you were angling for ExactSizeIterator::len? I think given that it doesn't exist for RangeInclusive<usize> due to overflow we may as well write it manually like:

I was trying to compute the length based upon the range, mainly as a defensive "one source of truth" strategy (i.e., can't change the range and forget to change the length).

Reply:
I don't really mind either way here. If you would prefer to define the range later then I'm fine to change it back.

Reply:
Nah that's fine, lets leave it